### PR TITLE
Refactor /bin/run-nextclade into nextclade.smk

### PR DESCRIPTION
**Pull request by corneliusroemer**
Since we now run Nextclade twice, once with normal dataset
and once with 21L dataset, it's important to keep datasets
separate.

The ./bin/run-nextclade does not allow concurrent running
as it will for example download nextclade twice and download

It's cleaner to turn the legacy shell script into a few
snakemake rules, this is the general way forward for ingest,
anyways

Local tests with config/debug_sample_genbank.yaml passed
This is in any case better than the current buggy solution

Test runs:
- [x] Genbank ingest on branch:  `62f54a4b-7b7b-469d-91d5-601f33d08d75`
- [x] GISAID ingest on branch: `4444f3ee-b50d-4fce-9bba-d8cf667c7e46` -> failed only due to later error

### CODE REVIEW

**Comment by ivan-aksamentov** [comment]
I don't see any obvious defects in the diff, but I am also not in the loop about the recent developments. In particular, where this is called, especially twice. But so far LGTM.

**Review by ivan-aksamentov — APPROVED** [review]

**Comment by ivan-aksamentov** [comment]
This sounds like something Nextclade should be doing, instead of hacking it in Snakemake, and throwing away alignment and stuff :)
https://github.com/nextstrain/ncov-ingest/blob/aa96fbdf513adfcf9a2d0f7d0f8a0cffec8dfd35/workflow/snakemake_rules/nextclade.smk#L5-L8
I am not sure what needs to be done, but feel free to open a conversation on Slack and/or GitHub.

File: workflow/snakemake_rules/nextclade.smk [hunk 1]
```diff
@@ -102,6 +102,37 @@ rule get_sequences_without_nextclade_annotations:
         echo "[ INFO] Number of {wildcards.reference} sequences to run Nextclade on: $(grep -c '^>' {output.fasta})"
         """
 
+rule download_nextclade_executable:
```
Comment (joverlee521) [inline workflow/snakemake_rules/nextclade.smk:105]: Do we still need to do this even though Nextclade is part of the nextstrain/base Docker image? Is this to ensure we always have the latest version of Nextclade in our ncov-ingest runs?

File: workflow/snakemake_rules/nextclade.smk [hunk 2]
```diff
@@ -102,6 +102,37 @@ rule get_sequences_without_nextclade_annotations:
...
+    input: "nextclade"
+    output:
+        dataset = "data/nextclade_data/{dataset_name}.zip"
+    shell:
+        """
+        ./nextclade dataset get --name="{wildcards.dataset_name}" --output-zip={output.dataset} --verbose
+        """
 
 GENES = "E,M,N,ORF1a,ORF1b,ORF3a,ORF6,ORF7a,ORF7b,ORF8,ORF9b,S"
 GENES_SPACE_DELIMITED = GENES.replace(",", " ")
```
[the remark this answers is not in the document]
Comment (corneliusroemer) [reply workflow/snakemake_rules/nextclade.smk:138]: Should probably do next time I touch Nextclade - this doesn't harm as is 🙃

File: workflow/snakemake_rules/nextclade.smk (hunk 1 above)
Comment (corneliusroemer) [reply workflow/snakemake_rules/nextclade.smk:105]: Not needed, but also doesn't do harm as far as I see it. We can remove it later. Will create an issue to remove in future.

**Pull request merged**